cluster/src/main/java/com/hartwig/pipeline/transfer/SbpFileApiUpdate.java

```diff
@@ -59,14 +62,26 @@ public void accept(final Blob blob) {
 
                 for (ApiFileOperation fileOperation : fileOperations) {
                     if (fileOperation.path().equals(blob.getName().substring(blob.getName().indexOf("/") + 1))) {
-                        LOGGER.info("Applying: {}", fileOperation);
+                        LOGGER.info("Applying [{}] for [{}]", fileOperation, blob.getName());
                         fileOperation.apply(sbpApi, fileResponse);
                     }
                 }
             }
         }
     }
 
+    private void storeOperations(List<StageOutput> stageOutputs) {
```

Comment:
I would suggest reverting all this to get back to a place where you can duplicate the bug. I don't see a possible concurrency issue as the new code only processes the list which should already contain the duplicate. If you can't duplicate it after that, it suggests we still don't quite understand the problem.

Reply:
I don't think we should keep this code, its a little confusing when you don't have the full context of why its here. If we do keep, the second info statement should probably be a WARN.